loki_client.go

```diff
@@ -62,6 +65,9 @@ type LokiClient struct {
 
 // Handle handles adding a new label set and a message to the batch
 func (m *LokiClient) Handle(ls model.LabelSet, t time.Time, s string) error {
+	if m.logWrapper.MaxErrors != 0 && len(m.logWrapper.errors) >= m.logWrapper.MaxErrors {
```

Comment:
wdyt about using `0` to signal lack of retries and returning err on first error and `-1` to disable it?